### ISSUE TITLE
Losses appear as nan in the middle of training

### ISSUE THREAD

**Issue by mattans** [facebookresearch/maskrcnn-benchmark#295]
## 🐛 Bug

Losses appear as nan in the middle of training

> index created!
> 2018-12-20 20:53:27,127 maskrcnn_benchmark.trainer INFO: Start training
> 2018-12-20 20:54:29,211 maskrcnn_benchmark.trainer INFO: eta: 3 days, 5:35:02  iter: 20  loss: 1.2211 (1.5259)  loss_classifier: 0.3178 (0.5123)  loss_box_reg: 0.0390 (0.0496)  loss_mask: 0.7223 (0.7687)  loss_objectness: 0.0829 (0.1821)  loss_rpn_box_reg: 0.0074 (0.0131)  time: 2.7855 (3.1041)  data: 0.0030 (0.2847)  lr: 0.007173  max mem: 2348
> 2018-12-20 20:55:29,711 maskrcnn_benchmark.trainer INFO: eta: 3 days, 4:34:45  iter: 40  loss: 0.9601 (1.2970)  loss_classifier: 0.2422 (0.4144)  loss_box_reg: 0.0942 (0.0822)  loss_mask: 0.5431 (0.6802)  loss_objectness: 0.0304 (0.1088)  loss_rpn_box_reg: 0.0065 (0.0114)  time: 3.0272 (3.0645)  data: 0.0030 (0.1442)  lr: 0.007707  max mem: 2421
> 2018-12-20 20:56:32,402 maskrcnn_benchmark.trainer INFO: eta: 3 days, 5:08:44  iter: 60  loss: 1.0016 (1.2237)  loss_classifier: 0.2760 (0.3928)  loss_box_reg: 0.0901 (0.0862)  loss_mask: 0.5242 (0.6391)  loss_objectness: 0.0445 (0.0937)  loss_rpn_box_reg: 0.0096 (0.0120)  time: 3.1236 (3.0879)  data: 0.0030 (0.0973)  lr: 0.008240  max mem: 2486
> 2018-12-20 20:57:39,352 maskrcnn_benchmark.trainer INFO: eta: 3 days, 6:44:58  iter: 80  loss: 0.5626 (1.0964)  loss_classifier: 0.1681 (0.3451)  loss_box_reg: 0.0899 (0.0914)  loss_mask: 0.3550 (0.5695)  loss_objectness: 0.0305 (0.0791)  loss_rpn_box_reg: 0.0066 (0.0114)  time: 3.3448 (3.1528)  data: 0.0030 (0.0739)  lr: 0.008773  max mem: 2562
> 2018-12-20 20:58:52,110 maskrcnn_benchmark.trainer INFO: eta: 3 days, 9:09:17  iter: 100  loss: 0.6952 (1.0156)  loss_classifier: 0.2484 (0.3189)  loss_box_reg: 0.1446 (0.0989)  loss_mask: 0.2361 (0.5181)  loss_objectness: 0.0210 (0.0688)  loss_rpn_box_reg: 0.0061 (0.0109)  time: 3.6240 (3.2498)  data: 0.0030 (0.0598)  lr: 0.009307  max mem: 2562
> 2018-12-20 21:00:09,149 maskrcnn_benchmark.trainer INFO: eta: 3 days, 11:38:32  iter: 120  loss: 0.6549 (0.9711)  loss_classifier: 0.2062 (0.3094)  loss_box_reg: 0.1184 (0.1038)  loss_mask: 0.2845 (0.4868)  loss_objectness: 0.0163 (0.0609)  loss_rpn_box_reg: 0.0068 (0.0103)  time: 3.7290 (3.3502)  data: 0.0030 (0.0505)  lr: 0.009840  max mem: 2562
> 2018-12-20 21:01:30,367 maskrcnn_benchmark.trainer INFO: eta: 3 days, 14:09:28  iter: 140  loss: 0.9763 (1.2557)  loss_classifier: 0.4051 (0.5879)  loss_box_reg: 0.0841 (0.1133)  loss_mask: 0.3714 (0.4849)  loss_objectness: 0.0236 (0.0594)  loss_rpn_box_reg: 0.0095 (0.0102)  time: 4.0679 (3.4517)  data: 0.0040 (0.0438)  lr: 0.010373  max mem: 2562
> 2018-12-20 21:02:32,856 maskrcnn_benchmark.trainer INFO: eta: 3 days, 13:07:05  iter: 160  loss: nan (nan)  loss_classifier: nan (nan)  loss_box_reg: nan (nan)  loss_mask: nan (nan)  loss_objectness: nan (nan)  loss_rpn_box_reg: 0.0074 (nan)  time: 2.6966 (3.4108)  data: 0.0030 (0.0388)  lr: 0.010907  max mem: 2562
> 2018-12-20 21:03:22,665 maskrcnn_benchmark.trainer INFO: eta: 3 days, 10:32:52  iter: 180  loss: nan (nan)  loss_classifier: nan (nan)  loss_box_reg: nan (nan)  loss_mask: nan (nan)  loss_objectness: nan (nan)  loss_rpn_box_reg: nan (nan)  time: 2.4793 (3.3085)  data: 0.0030 (0.0349)  lr: 0.011440  max mem: 2562
> 2018-12-20 21:04:11,928 maskrcnn_benchmark.trainer INFO: eta: 3 days, 8:25:14  iter: 200  loss: nan (nan)  loss_classifier: nan (nan)  loss_box_reg: nan (nan)  loss_mask: nan (nan)  loss_objectness: nan (nan)  loss_rpn_box_reg: nan (nan)  time: 2.4598 (3.2240)  data: 0.0040 (0.0318)  lr: 0.011973  max mem: 2562
> 2018-12-20 21:05:01,980 maskrcnn_benchmark.trainer INFO: eta: 3 days, 6:46:02  iter: 220  loss: nan (nan)  loss_classifier: nan (nan)  loss_box_reg: nan (nan)  loss_mask: nan (nan)  loss_objectness: nan (nan)  loss_rpn_box_reg: nan (nan)  time: 2.4877 (3.1584)  data: 0.0040 (0.0293)  lr: 0.012507  max mem: 2562
> 2018-12-20 21:05:51,639 maskrcnn_benchmark.trainer INFO: eta: 3 days, 5:20:46  iter: 240  loss: nan (nan)  loss_classifier: nan (nan)  loss_box_reg: nan (nan)  loss_mask: nan (nan)  loss_objectness: nan (nan)  loss_rpn_box_reg: nan (nan)  time: 2.4717 (3.1021)  data: 0.0040 (0.0273)  lr: 0.013040  max mem: 2562
> 2018-12-20 21:06:40,451 maskrcnn_benchmark.trainer INFO: eta: 3 days, 4:03:37  iter: 260  loss: nan (nan)  loss_classifier: nan (nan)  loss_box_reg: nan (nan)  loss_mask: nan (nan)  loss_objectness: nan (nan)  loss_rpn_box_reg: nan (nan)  time: 2.4390 (3.0512)  data: 0.0030 (0.0254)  lr: 0.013573  max mem: 2562
> 2018-12-20 21:07:29,679 maskrcnn_benchmark.trainer INFO: eta: 3 days, 2:59:37  iter: 280  loss: nan (nan)  loss_classifier: nan (nan)  loss_box_reg: nan (nan)  loss_mask: nan (nan)  loss_objectness: nan (nan)  loss_rpn_box_reg: nan (nan)  time: 2.4456 (3.0091)  data: 0.0030 (0.0239)  lr: 0.014107  max mem: 2562
> 2018-12-20 21:08:18,764 maskrcnn_benchmark.trainer INFO: eta: 3 days, 2:03:18  iter: 300  loss: nan (nan)  loss_classifier: nan (nan)  loss_box_reg: nan (nan)  loss_mask: nan (nan)  loss_objectness: nan (nan)  loss_rpn_box_reg: nan (nan)  time: 2.4386 (2.9721)  data: 0.0039 (0.0226)  lr: 0.014640  max mem: 2562
> 2018-12-20 21:09:07,970 maskrcnn_benchmark.trainer INFO: eta: 3 days, 1:14:30  iter: 320  loss: nan (nan)  loss_classifier: nan (nan)  loss_box_reg: nan (nan)  loss_mask: nan (nan)  loss_objectness: nan (nan)  loss_rpn_box_reg: nan (nan)  time: 2.4556 (2.9401)  data: 0.0030 (0.0214)  lr: 0.015173  max mem: 2562
> 2018-12-20 21:09:58,525 maskrcnn_benchmark.trainer INFO: eta: 3 days, 0:37:16  iter: 340  loss: nan (nan)  loss_classifier: nan (nan)  loss_box_reg: nan (nan)  loss_mask: nan (nan)  loss_objectness: nan (nan)  loss_rpn_box_reg: nan (nan)  time: 2.4938 (2.9159)  data: 0.0040 (0.0204)  lr: 0.015707  max mem: 2562
> 2018-12-20 21:10:49,286 maskrcnn_benchmark.trainer INFO: eta: 3 days, 0:04:56  iter: 360  loss: nan (nan)  loss_classifier: nan (nan)  loss_box_reg: nan (nan)  loss_mask: nan (nan)  loss_objectness: nan (nan)  loss_rpn_box_reg: nan (nan)  time: 2.4881 (2.8949)  data: 0.0050 (0.0196)  lr: 0.016240  max mem: 2562

## To Reproduce

I am running my own dataset, with coco-like api. 180 images. Same dataset is used for training and testing. 

YAML file:

> MODEL:
>   META_ARCHITECTURE: "GeneralizedRCNN"
>   WEIGHT: "catalog://ImageNetPretrained/MSRA/R-50"
>   BACKBONE:
>     CONV_BODY: "R-50-FPN"
>     OUT_CHANNELS: 256
>   RPN:
>     USE_FPN: True
>     ANCHOR_STRIDE: (4, 8, 16, 32, 64)
>     PRE_NMS_TOP_N_TRAIN: 2000
>     PRE_NMS_TOP_N_TEST: 1000
>     POST_NMS_TOP_N_TEST: 1000
>     FPN_POST_NMS_TOP_N_TEST: 1000
>   ROI_HEADS:
>     USE_FPN: True
>   ROI_BOX_HEAD:
>     POOLER_RESOLUTION: 7
>     POOLER_SCALES: (0.25, 0.125, 0.0625, 0.03125)
>     POOLER_SAMPLING_RATIO: 2
>     FEATURE_EXTRACTOR: "FPN2MLPFeatureExtractor"
>     PREDICTOR: "FPNPredictor"
>   ROI_MASK_HEAD:
>     POOLER_SCALES: (0.25, 0.125, 0.0625, 0.03125)
>     FEATURE_EXTRACTOR: "MaskRCNNFPNFeatureExtractor"
>     PREDICTOR: "MaskRCNNC4Predictor"
>     POOLER_RESOLUTION: 14
>     POOLER_SAMPLING_RATIO: 2
>     RESOLUTION: 28
>     SHARE_BOX_FEATURE_EXTRACTOR: False
>   MASK_ON: True
> DATASETS:
>   TRAIN: ("tau19", "tau19")
>   TEST: ("tau19",)
> DATALOADER:
>   SIZE_DIVISIBILITY: 32
> SOLVER:
>   BASE_LR: 0.02
>   WEIGHT_DECAY: 0.0001
>   STEPS: (60000, 80000)
>   MAX_ITER: 90000
>   IMS_PER_BATCH: 2
> 

## Environment

PyTorch version: 1.0.0
Is debug build: No
CUDA used to build PyTorch: 10.0

OS: ??Microsoft Windows 10 Education
GCC version: Could not collect
CMake version: Could not collect

Python version: 3.7
Is CUDA available: Yes
CUDA runtime version: 10.0.130
GPU models and configuration: GPU 0: GeForce MX150
Nvidia driver version: 411.31
cuDNN version: Could not collect

Versions of relevant libraries:
[pip] Could not collect
[conda] Could not collect

**Comment by fmassa** [comment]
Hi,

This can happen for a number of reasons. The simplest reason can be that the learning rate is too large and training diverges.

And this indeed seems to be the reason, as you've kept the original learning rate (0.02) but for a single GPU (batch size 2). You should adapt the learning rate according to https://github.com/facebookresearch/maskrcnn-benchmark#single-gpu-training
